repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -1994,14 +1995,27 @@ private void addCategoriesToTermEntity(AttributeMutationContext ctx, List<Object
         RequestContext.get().endMetricRecord(metricRecorder);
     }
 
-    private void addMeaningsToEntity(AttributeMutationContext ctx, List<Object> newElementsCreated) {
+    private void addMeaningsToEntity(AttributeMutationContext ctx, List<Object> createdElements, List<AtlasEdge> deletedElements) {
         MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("addMeaningsToEntity");
         // handle __terms attribute of entity
-        List<AtlasVertex> meanings = newElementsCreated.stream().map(x -> ((AtlasEdge) x).getOutVertex()).collect(Collectors.toList());
+        List<AtlasVertex> meanings = createdElements.stream()
+                .map(x -> ((AtlasEdge) x).getOutVertex())
+                .filter(x -> !ENTITY_DELETED_STATUS.equals(x.getProperty(STATE_PROPERTY_KEY, String.class)))
```

Comment:
Inverse Condition -> check if "is ACTIVE" instead "not DELETED"

Also, why not allowing deleted skipping DELETED terms?
Let us discuss this in sync tomorrow

Reply:
Yes I should add ACTIVE check, because PURGED entity must not get added, fixing those.